apps/web/src/components/Onboarding/ImportWallet/SeedPhraseTab.tsx

```diff
@@ -0,0 +1,156 @@
+      const words = mnemonic.split(" ");
+      if (!MNEMONIC_SIZE_OPTIONS.includes(words.length)) {
+        throw new Error(`the mnemonic must be ${MNEMONIC_SIZE_OPTIONS.join(", ")} words long`);
+      }
+      words.slice(0, mnemonicSize).forEach((word, i) => update(i, { val: word }));
+      return Promise.resolve();
+    });
+
+  const onSubmit = ({ mnemonic }: FormValues) =>
+    handleAsyncAction(async () => {
+      if (!validateMnemonic(mnemonic.map(({ val }) => val).join(" "))) {
+        throw new Error("Invalid Mnemonic");
+      }
+      return openWith(<SetupPassword mode="mnemonic" />);
+    });
+
+  const clearAll = () =>
+    form.setValue(
+      "mnemonic",
+      range(mnemonicSize).map(() => ({ val: "" }))
+    );
+
+  return (
+    <FormProvider {...form}>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <Flex flexDirection="column" gap="30px">
+          <Accordion allowToggle>
+            <AccordionItem>
+              <AccordionButton justifyContent="center" gap="10px" color={color("900")}>
+                <Heading size="md">{mnemonicSize} word seed phrase</Heading>
+                <AccordionIcon />
+              </AccordionButton>
+
+              <AccordionPanel>
+                <Flex gap="8px">
+                  <RadioButtons
+                    inputName="mnemonicSize"
+                    onSelect={changeMnemonicSize}
+                    options={MNEMONIC_SIZE_OPTIONS}
+                  />
+                </Flex>
+              </AccordionPanel>
+            </AccordionItem>
+          </Accordion>
+
+          <Grid gridRowGap="16px" gridColumnGap="12px" gridTemplateColumns="repeat(3, 1fr)">
+            {fields.map((field, index) => (
+              <GridItem key={field.id}>
+                <Text
+                  position="absolute"
+                  zIndex={1}
+                  marginTop={{ lg: "10px", base: "8px" }}
+                  marginLeft={{ lg: "16px", base: "10px" }}
+                  color={color("white", "black")}
+                  textAlign="right"
+                  size={{ lg: "lg", base: "xs" }}
+                >
+                  {String(index + 1).padStart(2, "0")}.
+                </Text>
+                <MnemonicAutocomplete
+                  inputName={`mnemonic.${index}.val`}
+                  inputProps={{
+                    ...register(`mnemonic.${index}.val`, { required: true }),
+                    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+                    onPaste: async e => {
+                      e.preventDefault();
+                      return pasteMnemonic(e.clipboardData.getData("text/plain"));
+                    },
+                    variant: "mnemonic",
+                    placeholder: `word #${index + 1}`,
+                  }}
+                />
+              </GridItem>
+            ))}
+          </Grid>
+          <Button gap="4px" onClick={clearAll} variant="ghost">
+            <Icon as={CloseIcon} />
```

Comment:
Wouldn't the Button `leftIcon` prop be more appropriate for placing a CloseIcon?

Reply:
sounds reasonable. might need some theme adjustments though because its default margin is 8px, but we need only 4.. let's use it in our next PRs